hydra-plutus/src/Hydra/Contract/Head.hs

```diff
@@ -357,7 +359,9 @@ checkContest ctx contestationDeadline parties closedSnapshotNumber sig headId =
     && mustBeSignedByParticipant ctx headId
     && mustBeWithinContestationPeriod
     && hasST headId outValue
+    && mustPreserveValue outValue headOutValue
  where
+  headOutValue = txOutValue . head $ txInfoOutputs txInfo
   outValue =
     maybe mempty (txOutValue . txInInfoResolved) $ findOwnInput ctx
```

Comment:
```suggestion
    && mustPreserveValue
 where 
  mustPreserveValue =
    traceIfFalse "head value is not preserved" $
      val == val'
  
  val' = txOutValue . head $ txInfoOutputs txInfo
  
  val = maybe mempty (txOutValue . txInInfoResolved) $ findOwnInput ctx
```

Could: follow a bit closer the terminology of the paper. Also, functions don't really help here when they are getting inlined anways (IMO).